nixos/modules/services/desktops/pipewire/wireplumber.nix

```diff
@@ -4,6 +4,42 @@ let
   pwCfg = config.services.pipewire;
   cfg = pwCfg.wireplumber;
   pwUsedForAudio = pwCfg.audio.enable;
+  configUnwantedUnsettingWarning = ''
+    When setting attributes here, make sure to copy the default attributes
+    along up to the top-level attribute that you're modifying. Otherwise
+    you'll unset all other attributes. This means that if you want to change
+    `log.level` in the following:
+    ```
+      context.properties = {
+        log.level = 2
+        wireplumber.script-engine = lua-scripting
+      }
+    ```
+    if you only add
+    ```
+      context.properties = {
+        log.level = 2
+      }
+    ```
+    to your extra config, you would unset `wireplumber.script-engine`.
+    If you don't want to accidentally unset sub-attributes in the same top-level
+    attribute like this, then be sure to copy the defaults for the top-level
+    attribute (in this case `context.properties`).
+  '';
+  mkConfigFileFromDefault = name: extraConfig:
+    {
+      "share/wireplumber/${name}.conf" = pkgs.writeText name (
```

Comment:
Does wireplumber not work with drop-ins? Or is there a major use case for unsetting options?

Reply:
As far as I can tell, WirePlumber does not merge `.conf` files at the moment and it also does not merge attribute objects in them. So if you redefine a top-level attribute, you basically unset all of its sub-attributes unless you copy them along. If my understanding of this is incorrect, then please correct me. This is just what I've come to believe based on some testing. This is sadly not documented well/at all it seems.

Reply:
I feel like that shouldn't be the case, but I'm going to go ask upstream.

Reply:
The way I tested this was to copy over `/usr/share/wireplumber/wireplumber.conf` into `/etc/wireplumber` and then setting `log.level = 4` in `context.properties` at the top and setting `context.properties = {}` at the bottom. Maybe the way to go is just to not copy `/usr/share/wireplumber/wireplumber.conf` and hope WP merges `/etc` and `/share` contents. Does wireplumber pick up the files from `/run/current-system/sw/share/wireplumber`?

Reply:
Setting things in one file might be weird, what if you set it in `/etc/wireplumber/wireplumber.conf.d/dropin.conf`? And no, it doesn't pick up /run.

Reply:
I think I've tried that already with it not picking that up, but I'll try that again. Maybe I just bodged something in my last attempt

Reply:
I had also searched the entire WP repo at 0.4 not being able to find anything on `conf.d` merging. I think that might've been added (or at least only documented) for 0.5

Reply:
Asahi audio also adds a file to `wireplumber.conf.d` to configure realtime settings, I somehow missed that. I guess they too assume `conf.d` merging works. I have tried your exact test with a `dropin.conf` in `wireplumber.conf.d` with the following contents:
```
context.properties = {
  log.level = 4
}
```
After a `systemctl --user daemon-reload` and a restart of `wireplumber.service` the log level has not changed though. That's what I meant. Perhaps you see some issue with how I'm testing this

Reply:
That definitely sounds like a bug.